arduino-ide-extension/src/browser/arduino-frontend-contribution.tsx

```diff
@@ -549,6 +551,37 @@ export class ArduinoFrontendContribution
     }
   }
 
+  onWillStop(): OnWillStopAction {
+    return {
+      reason: 'Temp Sketch',
+      action: async () => {
+        const sketch = await this.sketchServiceClient.currentSketch();
+        if (!sketch) {
+          return true;
+        }
+        const isTemp = await this.sketchService.isTemp(sketch);
+        if (!isTemp) {
+          return true;
+        }
+        const dialog = new TempSketchDialog();
```

Comment:
@msujew may I ask you why did you choose to implement a new Theia dialog instead of using a native dialog? I think the latter would be a better option for this kind of features.